contracts/settling_game/modules/food/Food.cairo

```diff
@@ -0,0 +1,542 @@
+# @param food_building_id: food building id
+@external
+func harvest{
+    syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr : BitwiseBuiltin*
+}(token_id : Uint256, harvest_type : felt, food_building_id : felt):
+    alloc_locals
+
+    # check id and harvest type
+    let (is_correct_id) = Food.assert_ids(food_building_id)
+    with_attr error_message("FOOD: Incorrect Building ID"):
+        assert_not_zero(is_correct_id)
+    end
+
+    let (is_correct_harvest_type) = Food.assert_harvest_type(harvest_type)
+    with_attr error_message("FOOD: Incorrect Harvest ID"):
+        assert_not_zero(is_correct_harvest_type)
+    end
+
+    # check owner
+    Module.ERC721_owner_check(token_id, ExternalContractIds.S_Realms)
+
+    # contracts
+    let (s_realms_address) = Module.get_external_contract_address(ExternalContractIds.S_Realms)
+    let (resources_address) = Module.get_external_contract_address(ExternalContractIds.Resources)
+    let (owner) = realms_IERC721.ownerOf(s_realms_address, token_id)
+
+    let (block_timestamp) = get_block_timestamp()
+
+    # build required building
+    if food_building_id == RealmBuildingsIds.Farm:
+        let (packed) = farms.read(token_id)
+        # unpack
+        let (unpacked_food_buildings : FoodBuildings) = Food.unpack_food_buildings(packed)
+        # calculate harvests
+        let (total_harvest, total_remaining, decayed_farms) = Food.calculate_harvest(
+            unpacked_food_buildings.UpdateTime, block_timestamp
+        )
+        # save tempvar of unpacked number
+        tempvar number_built = unpacked_food_buildings.NumberBuilt
+        tempvar current_harvests = unpacked_food_buildings.CollectionsLeft
+
+        # builtins
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+        tempvar bitwise_ptr = bitwise_ptr
+    else:
+        let (packed) = fishing_villages.read(token_id)
+        # unpack
+        let (unpacked_food_buildings : FoodBuildings) = Food.unpack_food_buildings(packed)
+        # calculate harvests
+        let (total_harvest, total_remaining, decayed_farms) = Food.calculate_harvest(
+            unpacked_food_buildings.UpdateTime, block_timestamp
+        )
+        # save tempvar of unpacked number
+        tempvar number_built = unpacked_food_buildings.NumberBuilt
+        tempvar current_harvests = unpacked_food_buildings.CollectionsLeft
+
+        # builtins
+        tempvar syscall_ptr = syscall_ptr
+        tempvar range_check_ptr = range_check_ptr
+        tempvar pedersen_ptr = pedersen_ptr
+        tempvar bitwise_ptr = bitwise_ptr
+    end
+
+    # total food to harvest
+    let total_food = total_harvest * BASE_FOOD_PRODUCTION
+
+    # update storage
+    update(current_harvests, number_built, total_harvest, decayed_farms, token_id, food_building_id)
```

Comment:
Is third param here correct? Should it be`total_harvest` or `total_food`? Asking because `total_food` is calculated just above.

Reply:
yes this is correct - total_food is what is to be minted as tokens or stored in the storehouse. total harvests are how many of the harvests the player is collecting.